mne/time_frequency/tfr.py

```diff
@@ -1144,9 +1205,355 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+            tfr.pick_channels(pick_names)
+
+        if exclude == 'bads':
+            exclude = [ch for ch in tfr.info['bads']
+                       if ch in tfr.info['ch_names']]
+        if exclude is not None:
+            tfr.drop_channels(exclude)
+
+        info = tfr.info
+        data_types = {'eeg', 'grad', 'mag', 'seeg', 'ecog', 'hbo', 'hbr'}
+        ch_types = set(ch_type for ch_type in data_types if ch_type in self)
+
+        # if multiple sensor types: one plot per channel type, recursive call
+        if len(ch_types) > 1:
+            figs = list()
+            for this_type in ch_types:  # pick corresponding channel type
+                tf_ = tfr.copy().pick_channels(
+                    [info['ch_names'][idx] for idx in range(info['nchan'])
+                     if channel_type(info, idx) == this_type])
+                if _check_multiple_data_channel_types(tf_.info):
+                    raise RuntimeError(
+                        'Possibly infinite loop due to channel selection '
+                        'problem. This should never happen! Please check '
+                        'your channel types.')
+                figs.append(
+                    tf_.plot_joint(
+                        timefreqs=timefreqs, picks=None, baseline=baseline,
+                        mode=mode, tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                        vmin=vmin, vmax=vmax, cmap=cmap, dB=dB,
+                        colorbar=colorbar, show=False, title=title,
+                        layout=layout, yscale=yscale, combine=combine,
+                        exclude=None, topomap_args=topomap_args,
+                        verbose=verbose))
+            return figs
+
+        ##############
+        # Image plot #
+        ##############
+
+        # prepare for image plot
+        timefreq_error_msg = (
+            "Supplied `timefreqs` are somehow malformed. Please supply None, "
+            "a list of tuple pairs, or a dict of such tuple pairs, not ")
+        if timefreqs is None:
+            n_timefreqs = 1
+        else:
+            try:
+                n_timefreqs = len(timefreqs)
+            except TypeError:
+                raise ValueError(timefreq_error_msg + str(type(timefreqs)))
+
+        fig, tf_ax, map_ax, cbar_ax = _prepare_joint_axes(n_timefreqs)
+
+        cmap = _setup_cmap(cmap)
+
+        # image plot
+        fig, tfr = tfr._plot(
+            picks=None, baseline=baseline, mode=mode, tmin=tmin, tmax=tmax,
+            fmin=fmin, fmax=fmax, vmin=vmin, vmax=vmax, cmap=cmap, dB=dB,
+            colorbar=False, show=False, title=title, axes=tf_ax,
+            layout=layout, yscale=yscale, combine=combine,
+            exclude=None, copy=False, verbose=verbose)
+
+        ####################
+        # Handle timefreqs #
+        ####################
+
+        # find and/or setup timefreqs
+        try:  # There are a trillin ways for this to go wrong ...
```

Comment:
@agramfort are you ok with this try/except?

Reply:
not really. You shoudl have a simple private auxilary function that does the checks on types, dims and values.

Reply:
Here is one error case I'm not sure about how to check without a try: if `timefreqs` is a non-iterable. There's a ton of iterables. I could do `hasattr(item, '__len__')`, but that does not seem perfect.